### PR TITLE
Have a dedicated neighborhood property and a get_neighborhood method on Cell

### PR DESCRIPTION
**Pull request by quaquel**
This picks up on a discussion in #2296. In the current code, we have method named `Cell.neighborhood`. This is an unfortunate name. So we agreed to split this into a property for the direct neighborhood (i.e. radius = 1) and a `get_neighborhood` method for larger radii. We had a bit of discussion on the names, but I decided to keep it simple: `neighborhood` and `get_neighborhood`. I have updated the docstrings to clearly explain when to use which one.

### CODE REVIEW

File: mesa/experimental/cell_space/cell.py [hunk 1]
```diff
@@ -131,10 +131,34 @@ def is_full(self) -> bool:
...
 
+    @property
+    @cache
+    def neighborhood(self) -> CellCollection:
+        """Returns the direct neigborhood of the cell
+
+        This is equivalent to cell.get_neighborhood(radius=1)
+
+        """
+        return self.get_neighborhood(radius=1)
```
Comment (EwoutH) [inline mesa/experimental/cell_space/cell.py:142]: The `radius=1` is redundant right? I think it's more powerful if it uses the fully default get_neighborhood() method and is thus equivalent at calling that.
```suggestion
        return self.get_neighborhood()
```

**Review by EwoutH — COMMENTED** [review]
I like the general design!

Any performance implication on caching? Mostly good probably, right?

**Comment by quaquel** [comment]
just a quick observation about cellspaces. A random walk is now a one liner: `self.cell.neighborhood.select_random_cell()`.

**Comment by EwoutH** [comment]
That’s really nice!

I would appreciate a deprecationwarning for `neighborhood()`

File: mesa/experimental/cell_space/cell.py [hunk 2]
```diff
@@ -131,10 +131,34 @@ def is_full(self) -> bool:
     def __repr__(self):  # noqa
         return f"Cell({self.coordinate}, {self.agents})"
 
+    @property
+    @cache
```
Comment (Corvince) [inline mesa/experimental/cell_space/cell.py:135]: You can use cachedproperty here (I think from functools)

**Review by Corvince — APPROVED** [review]

**Comment by quaquel** [comment]
> I would appreciate a deprecationwarning for neighborhood()

AFAIK that is not possible because the name clashes with the property.

**Comment by EwoutH** [comment]
It looks like this does introduce some errors.

**Comment by EwoutH** [comment]
These benchmark results are exactly opposite of https://github.com/projectmesa/mesa/pull/2308.

So seems to cancel out and are just based on an outdated branch or whatever.

**Review by EwoutH — APPROVED** [review]
Was starting to type something about API compatibility and deprecation warnings, but then remembered this is all in the experimental space.

Love the experimental space!

**Comment by EwoutH** [comment]
Thanks, great to have it in!

Please make sure that pre-commit passes before merging. Always a bitch to clean up (done in 7037a32).

**Pull request merged**